app/models/reservation.rb

```diff
@@ -25,7 +25,7 @@ class Reservation < ActiveRecord::Base
   validate :not_in_past, :available, :check_banned, on: :create
 
   # correctly update the overdue flag if necessary
-  before_save :update_overdue, if: :checked_out?
+  before_save :update_overdue, if: :checked_out?, unless: self.changes.empty?
```

Comment:
I got the compounded idea from the active record documentation. However it doesn't mention there if it's an "and" or an "or". So I"ll go with your suggestion and implement it in the call back itself